bspump/elasticsearch/sink.py

```diff
@@ -20,15 +20,29 @@ class ElasticSearchSink(Sink):
 	of the sink:
 
 	es_index (STRING): ElasticSearch index name
+
+	When specifying action: custom in the configuration,
+	custom_data_feeder coroutine needs to be provided,
+	which accepts items as its only parameter and yields data as Python generator.
+	The example implementation is:
+
+	async def my_data_feeder(items):
+		for _id, data in items:
+			yield b'{"create":{}}\n' if _id is None else orjson.dumps(
+				{"index": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
+			)
+			yield data
+
 	"""
 
 	ConfigDefaults = {
 		"index_prefix": "bspump_",  # Obsolete, use 'index'
 		"index": "bspump_",
+		'action': 'create',  # create, index, update, delete, custom
 	}
 
 
-	def __init__(self, app, pipeline, connection, id=None, config=None, bulk_class=ElasticSearchBulk):
+	def __init__(self, app, pipeline, connection, id=None, config=None, bulk_class=ElasticSearchBulk, custom_data_feeder=None):
```

Comment:
`custom_data_feeder` should be renamed to `data_feeder` - the default is the "original "create or index" feeder.

Then we provide alternative feeders that is to be passed as `data_feeder` argument here.